ui/src/components/imquery/AddPropertyDialog.vue

```diff
@@ -63,29 +82,60 @@ watch(
   }
 );
 
+watch(
+  () => cloneDeep(selectedProperty.value),
+  newValue => {
+    if (isObjectHasKeys(selectedProperty.value)) {
+      whereOrMatch.value = buildProperty(selectedProperty.value as any);
```

Comment:
selectedProperty is typed, why casting?

Reply:
two different TreeNode interfaces (one in im-library and one in primevue) - because they are different casting is needed